runtime/auth/aws-signing-common/common/test/aws/smithy/kotlin/runtime/auth/awssigning/PresignerTest.kt

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.smithy.kotlin.runtime.auth.awssigning
+
+import aws.smithy.kotlin.runtime.auth.awscredentials.Credentials
+import aws.smithy.kotlin.runtime.auth.awscredentials.CredentialsProvider
+import aws.smithy.kotlin.runtime.client.endpoints.Endpoint
+import aws.smithy.kotlin.runtime.http.Headers
+import aws.smithy.kotlin.runtime.http.operation.EndpointResolver
+import aws.smithy.kotlin.runtime.http.operation.ResolveEndpointRequest
+import aws.smithy.kotlin.runtime.http.request.HttpRequest
+import aws.smithy.kotlin.runtime.http.request.HttpRequestBuilder
+import aws.smithy.kotlin.runtime.http.request.url
+import aws.smithy.kotlin.runtime.net.Url
+import aws.smithy.kotlin.runtime.operation.ExecutionContext
+import aws.smithy.kotlin.runtime.util.Attributes
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+private const val NON_HTTPS_URL = "http://localhost:8080/path/to/resource?foo=bar"
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class PresignerTest {
+    @Test
+    fun testSignedUrlAllowsHttp() = testSigningUrl("http://localhost:8080/path/to/resource?foo=bar")
```

Comment:
**suggestion** When adding test coverage for a bug I usually try to add a comment with a link for posterity (yes you can get this through git/github history but it's a nice quick thing to understand a test when refactoring).

e.g.

```kotlin
// see awslabs/aws-sdk-kotlin#938
@Test
fun testSignedUrlAllowsHttp() ...
```